lib/TableUtils.lua

```diff
@@ -1,5 +1,13 @@
 local TableUtils = {}
 
+local function getIterator(source)
+	if type(source) == "function" then
+		return source
```

Comment:
what's the point of this?

Reply:
Okay I see it used in the tests below, but what if you pass some random function by mistake that isn't supposed to be used as an iterator?

Reply:
I can't see an easy way to assert that the function will behave sensibly as an iterator in a for loop. If it doesn't, the for loop will throw, which will provide a better error trace than if we try to make a proxied "safe" iterator that warns. Which I think is somewhat dangerous anyway.